libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -243,8 +243,6 @@ internal class MapboxTripSession(
     }
 
     private fun updateRawLocation(rawLocation: Location) {
-        if (state != TripSessionState.STARTED) return
-
```

Comment:
Can't say that I feel confident about this change. But I can't come-up with a reasonable scenario when this line could be necessary. If you know such a scenario please let me know so that I can write a test for it.

Reply:
@kmadsen Does that ring a bell? I remember we discussed this long time ago.

Reply:
Hmm would be good to find where it was added in github history. I was last to touch this code, but I was only copy pasting https://github.com/mapbox/mapbox-navigation-android/commit/824a80ba574d9f388313366cc07ef008f71131ad#diff-37365d0e69a56e8efddf9ff8e735c04b3474c7df666259edfbc961855c3556a2L590-L599

Reply:
Digging deeper, I'm also the one who added it 😅  haha.

https://github.com/mapbox/mapbox-navigation-android/commit/dafada2006df4af83171761df79076a70899a016#diff-37365d0e69a56e8efddf9ff8e735c04b3474c7df666259edfbc961855c3556a2R501-R503

But at least that PR shows exactly why it was added. https://github.com/mapbox/mapbox-navigation-android/pull/3919

Unit tests for race conditions are not very simple to implement, which explains why there isn't one.

Reply:
In the comments i linked this one https://github.com/mapbox/mapbox-navigation-android/pull/3920
There's some hacks to reproduce the issue within mapbox-navigation-android

Reply:
@VysotskiVadim did that additional context help?

Reply:
@kmadsen, as far I understand the idea is to:
1. set some route
2. set an empty route almost immediately via starting a coroutine in UI thread 
2. stop session immediately
3. check if there're any onRouteProgressChanged events

I tried it in the [vv-testing-race-conditions](https://github.com/mapbox/mapbox-navigation-android/compare/vv-handle-synchronous-location-in-mapbox-trip-session...vv-testing-race-conditions) branch. I didn't get any onRouteProgressChanged events 🙂 

I can't reproduce race condition _(TBH I'm not sure if we can use that term in the context of a single thread 🤔. We work only with UI thread here, don't we? 😅  )_ . I don't know if everything works fine because everything is fine or because I test in a wrong way. @kmadsen , can you please assist with testing? Did I test correctly in the **vv-testing-race-conditions** branch?

Reply:
> I tried it in the vv-testing-race-conditions branch. I didn't get any onRouteProgressChanged events 🙂

Thanks for testing!

Reply:
> We work only with UI thread here, don't we?

https://github.com/mapbox/mapbox-navigation-android/blob/dafada2006df4af83171761df79076a70899a016/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt#L92

It used to use the io dispatcher to communicate with nav-native. But think nav-native has been fixing these synchronization issues.

Seems to be the reason why this is no longer needed. 👍